### PR TITLE
Return an error instead of coercing unparseable Loki timestamps to Unix epoch

### LINKED ISSUE #49: Return an error instead of coercing unparseable Loki timestamps to Unix epoch
## Why

`entries_from_response` uses `.unwrap_or(0)` when parsing a Loki timestamp as `i64`. A timestamp of 0 means 1970-01-01 00:00:00 UTC — so a malformed Loki response silently produces entries that appear decades old rather than surfacing a data problem. Because the timestamp comes from the Loki API (an external system), a parse failure is an environmental error and should be returned, not silently substituted.

## Current state

`clients/src/loki.rs` line 86:

```rust
timestamp_ns: ts.parse().unwrap_or(0)
```

A parse failure produces timestamp 0, which flows into the `age` computation in `workflows/src/loki.rs:77` and causes the entry to sort as extremely old.

## Ideal state

The parse failure propagates as an error:

```rust
timestamp_ns: ts.parse().map_err(|_| LokiError::InvalidTimestamp(ts.to_owned()))?
```

A malformed Loki response surfaces immediately as a fetch error rather than distorting entry ages downstream. `LokiError` gains an `InvalidTimestamp(String)` variant to carry the offending value.

## Starting points

- `clients/src/loki.rs` — `entries_from_response`, line 86 (the `.unwrap_or(0)` call)
- `workflows/src/loki.rs:77` — downstream `age` computation that depends on a valid timestamp

## QA plan

1. Read `clients/src/loki.rs` — expect no `.unwrap_or(0)` on the timestamp parse; expect the function propagates a parse error via `?`
2. Check that `LokiError` has an `InvalidTimestamp` variant (or equivalent)
3. Run `cargo test` — expect all tests pass

## Done when

`entries_from_response` propagates a parse failure on Loki timestamps as an error instead of substituting 0.

### PR DESCRIPTION
**Pull request by ooloth**
Replaces `.unwrap_or(0)` on the Loki timestamp parse with `.with_context(|| format!("unparseable Loki timestamp: {ts}"))?` so that a malformed Loki response surfaces immediately as a fetch error rather than silently producing entries that appear decades old.

Changes:
- Changes `entries_from_response` to return `anyhow::Result<Vec<LogEntry>>`
- Uses `.with_context()` on the timestamp parse to produce a descriptive error
- Updates the call site in `entries` to propagate the error via `?`
- Surfaces Loki failures in the TUI status bar (via `StatusReport.errors`)
- Renders partial-refresh errors in CLI output
- Adds tests covering the error path and boundary values

Closes #49

### CODE REVIEW

File: clients/src/loki.rs [hunk 1]
```diff
@@ -72,20 +73,25 @@ pub async fn entries(
         .await
         .context("failed to parse Loki response")?;
 
-    Ok(entries_from_response(response))
+    entries_from_response(response)
 }
 
-fn entries_from_response(response: QueryRangeResponse) -> Vec<LogEntry> {
+fn entries_from_response(response: QueryRangeResponse) -> anyhow::Result<Vec<LogEntry>> {
```
[the remark this answers is not in the document]
Comment (ooloth) [reply clients/src/loki.rs:79]: Good catch — the PR description was stale. Updated it to reflect the actual implementation: we're using anyhow with `.with_context()` rather than a custom error enum, which aligns with the project's anyhow-only convention.

File: workflows/src/status.rs [hunk 2]
```diff
@@ -175,11 +175,10 @@ pub async fn run(params: StatusParams) -> Result<StatusReport> {
         errors.push("linear issues".to_string());
     }
 
-    // Loki errors are logged but don't fail the refresh (like the original behavior).
     for env in &params.loki_envs {
         match crate::loki::run(env).await {
             Ok(loki_items) => items.extend(loki_items.into_iter().map(StatusItem::Loki)),
-            Err(e) => eprintln!("loki ({} · {}): {e}", env.project, env.env),
+            Err(_) => errors.push(format!("loki ({} · {})", env.project, env.env)),
```
[the remark this answers is not in the document]
Comment (ooloth) [reply workflows/src/status.rs:181]: Valid point — the CLI was indeed losing visibility. Rather than adding `eprintln!` back just for Loki (which would be inconsistent with how GitHub and Linear failures are handled), I've added CLI rendering of `report.errors` for all sources. Now any unreachable source shows up in the CLI output under an "unreachable" section, same way the TUI shows it in the status bar.
